stubs/pyOpenSSL/METADATA.toml

```diff
@@ -1,2 +1,3 @@
 version = "23.0.*"
-requires = ["types-cryptography"]
+# Requires a version of cryptography with a `py.typed` file
+requires = ["cryptography>=35.0.0"]
```

Comment:
Interesting. I feel like it would make sense to enforce a minimal version for external dependencies of 3rd party stub.

Reply:
Definitely something to explore in the future!

Reply:
Without having to enforce it through CI. Maybe a mention in documentation that external dependencies should be at least pinned to their first `py.typed` version? (and optimally to the oldest version that has the types referenced by the stub)

Reply:
I'll try to work on docs as soon as this PR makes it over the line — this is a very good point, thanks!